qa/tasks/cephfs/mount.py

```diff
@@ -169,6 +169,26 @@ def assert_and_log_minimum_mount_details(self):
                      get_file(self.client_remote, self.client_keyring_path,
                               sudo=True).decode())
 
+    def is_stuck(self):
+        """
+        Check if mount is stuck/in a hanged state.
+        """
+        if not self.is_mounted():
+            return False
+
+        retval = self.client_remote.run(args=f'sudo stat {self.hostfs_mntpt}',
+                                        omit_sudo=False, wait=False).returncode
```

Comment:
this doesnt look right. Shouldn't you use `timeout` command to  wait?

Reply:
It was used in previous versions of this patch, `timeout` too would hang along with the rest of the command.

Reply:
Ah, crap. `/bin/timeout` does `waitpid()` for the child. My #47264 likely doesn't work (I am still testing it). How disappointing.

I think your approach can work but I'm confused why you fetch the return code in this line. Wouldn't that wait for the result?